typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -86,4 +186,610 @@ export class EvmIsmModule extends HyperlaneModule<
       config,
     });
   }
+
+  protected async updateRoutingIsm(params: {
+    destination: ChainName;
+    config: RoutingIsmConfig;
+    origin?: ChainName;
+    mailbox?: Address;
+    existingIsmAddress: Address;
+    logger: Logger;
+  }): Promise<IRoutingIsm> {
+    const { destination, config, mailbox, existingIsmAddress, logger } = params;
+
+    // 1. Filter for known domains
+    const { domains, safeConfigDomains } = EvmIsmModule.filterRoutingIsmDomains(
+      { config, multiProvider: this.multiProvider },
+    );
+    config.domains = domains;
+
+    // 2. Compute the delta
+    const delta: RoutingIsmDelta = await routingModuleDelta(
+      destination,
+      existingIsmAddress,
+      config,
+      this.multiProvider,
+      this.factories,
+      mailbox,
+    );
+
+    // Step 3: Get owner and signer
+    const signer = this.multiProvider.getSigner(destination);
+    const provider = this.multiProvider.getProvider(destination);
+    const owner = await DomainRoutingIsm__factory.connect(
+      existingIsmAddress,
+      provider,
+    ).owner();
+    const isOwner = eqAddress(await signer.getAddress(), owner);
+
+    if (existingIsmAddress && isOwner && !delta.mailbox) {
+      // 4. If possible, reconfigure existing ISM
+      return await this.reconfigureExistingRoutingIsm({
+        destination,
+        delta,
+        existingIsmAddress,
+        config,
+        mailbox,
+        logger,
+      });
+    } else {
+      // 5. Else, deploy a new set of routing ISM submodules
+      const submoduleAddresses = await this.deployRoutingIsmSubmodules({
+        destination,
+        config,
+        mailbox,
+      });
+
+      const owner = await resolveOrDeployAccountOwner(
+        this.multiProvider,
+        destination,
+        config.owner,
+      );
```

Comment:
hoping to remove this ICA resolution from these abstractions
we can resolve the local owner address at a higher level

Reply:
does that mean we should move the `transferOwnershipOfContracts` in here to update() as well?
<img width="571" alt="image" src="https://github.com/hyperlane-xyz/hyperlane-monorepo/assets/10051819/29dad35b-4c59-4cbd-a973-61b573f59f46">